assets/blocks/button/index.js

```diff
@@ -0,0 +1,75 @@
+import { __ } from '@wordpress/i18n';
+import { merge } from 'lodash';
+
+import './color-hooks';
+import { EditButtonBlock } from './edit-button';
+import { saveButtonBlock } from './save-button';
+
+/**
+ * Button block styles.
+ */
+export const BlockStyles = {
+	Fill: {
+		name: 'default',
+		label: __( 'Fill', 'sensei-lms' ),
+		isDefault: true,
+	},
+	Outline: {
+		name: 'outline',
+		label: __( 'Outline', 'sensei-lms' ),
+	},
+};
+
+/**
+ * Button block base.
+ *
+ * @param {Object} opts
+ * @param {Object} opts.settings Block settings.
+ */
+export const buttonBlock = ( { settings, ...options } ) => {
+	options = {
+		tagName: 'a',
+		...options,
+	};
```

Comment:
Settings are merged into block settings (like block name, title), the rest of options are passed to the edit and save functions. We can use this to make parts of the base component configurable/extendable. Right now it only has a `tagName` option for the output element.